### PR TITLE
ci: specific pipeline to run the benchmarking

### PR DESCRIPTION
**Pull request by v1v**
## Highlights
- Create a pipeline to be triggered only on a daily basis for the master branch. Will run the benchmark.
- Change multibranch pipeline to validate PRs but won't run on a daily basis and won't trigger the benchmark.
- Mutlibranch pipeline will validate the benchmark only for the branches.

## Workflows
- Single pipeline to validate the benchmark on daily basis for the master branch.
- Multibranch pipeline to validate each PR without running the benchmark tests. Based on GitHub push events.
- Multibranch pipeline to validate each branch in the origin by running the UTs and the benchmark. Based on GitHub push events.

## Test Cases
- Validated locally running the JJBB with the same docker imaged used by infra for the apm-ci, although we some tweaks to run the https service locally.

Depends on https://github.com/elastic/hey-apm/pull/128

### CODE REVIEW

File: .ci/scheduled-benchmark.groovy [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+pipeline {
+  agent any
+  environment {
+    BASE_DIR="src/github.com/elastic/hey-apm"
+    JOB_GIT_CREDENTIALS = "f6c7695a-671e-4f4f-a331-acdce44ff9ba"
+    GO_VERSION = "${params.GO_VERSION}"
+    STACK_VERSION = "${params.STACK_VERSION}"
+    NOTIFY_TO = credentials('notify-to')
+    JOB_GCS_BUCKET = credentials('gcs-bucket')
+    BENCHMARK_SECRET  = 'secret/apm-team/ci/apm-server-benchmark-cloud'
```
Comment (jalvz) [inline .ci/scheduled-benchmark.groovy]: not `java-agent-benchmark-cloud`?

File: .ci/scheduled-benchmark.groovy [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  options {
+    timeout(time: 1, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+  }
+  triggers {
+    cron('H H(3-5) * * 1-5')
```
Comment (jalvz) [inline .ci/scheduled-benchmark.groovy]: why is this duplicated? (https://github.com/elastic/hey-apm/pull/129/files#diff-90352688c666dc1a6ce60b25c2422b58R22)

File: .ci/scheduled-benchmark.groovy [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  }
+  parameters {
+    string(name: 'GO_VERSION', defaultValue: "1.12.1", description: "Go version to use.")
+    string(name: 'STACK_VERSION', defaultValue: "7.3.0-SNAPSHOT", description: "Stack version Git branch/tag to use")
```
Comment (jalvz) [inline .ci/scheduled-benchmark.groovy]: as you suggested offline, you can get the default version with something like 
`curl https://raw.githubusercontent.com/elastic/apm-server/master/vendor/github.com/elastic/beats/libbeat/version/version.go | grep defaultBeatVersion | cut -d'=' -f2`

File: .ci/scheduled-benchmark.groovy (hunk 1 above)
Comment (v1v) [reply .ci/scheduled-benchmark.groovy]: Already rebased with the previous PR

File: .ci/scheduled-benchmark.groovy [hunk 4]
```diff
@@ -119,3 +125,7 @@ pipeline {
     }
   }
 }
+
+def getVersion() {
+  return sh(script: """curl -s ${VERSION_FILE}  | grep defaultBeatVersion | cut -d'=' -f2 | sed 's#"##g'""", returnStdout: true).trim()
```
Comment (jalvz) [inline .ci/scheduled-benchmark.groovy]: probably needs an `-SNAPHOST` at the end, right? 

i'd also log out the version if doesn't happen already somewhere else

File: .ci/scheduled-benchmark.groovy (hunk 2 above)
Comment (v1v) [reply .ci/scheduled-benchmark.groovy]: That's a bit tricky, IIRC, it's required twice as one is creating the UI item and the other one is the pipeline implementation. The JJBB and pipeline are a bit tricky together.

So furthermore, the UI does require to be created always, and the pipeline is required to be run once to apply its changes, therefore, to enable that cron feature automatically we need to set the trigger in the UI too. It's weird but the pipeline object is not created in the CI jenkins unless there is build linked to it.

File: .ci/scheduled-benchmark.groovy (hunk 4 above)
Comment (v1v) [reply .ci/scheduled-benchmark.groovy]: aha, that's indeed a good point

**Pull request merged**